src/lease-process/lease.module.ts

```diff
@@ -0,0 +1,78 @@
+import { ActivityModule } from "../activity";
+import { Agreement, DraftOfferProposalPool, MarketModule } from "../market";
+import { NetworkModule } from "../network";
+import { Allocation, PaymentModule } from "../payment";
+import { StorageProvider } from "../shared/storage";
+import { Logger } from "../shared/utils";
+import { LeaseProcess, LeaseProcessOptions } from "./lease-process";
+import { LeaseProcessPool, LeaseProcessPoolOptions } from "./lease-process-pool";
+
+export interface LeaseModule {
+  /**
+   * Factory that creates a new lease process that's fully configured.
+   * This method will also create and start the payment process for the agreement.
+   *
+   */
+  createLease(agreement: Agreement, allocation: Allocation, options?: LeaseProcessOptions): LeaseProcess;
+  /**
+   * Factory that creates new lease process pool that's fully configured
+   */
+  createLeaseProcessPool(
+    draftPool: DraftOfferProposalPool,
+    allocation: Allocation,
+    options?: LeaseProcessPoolOptions,
+  ): LeaseProcessPool;
+}
+
+export class LeaseModuleImpl implements LeaseModule {
+  constructor(
+    private readonly deps: {
+      marketModule: MarketModule;
+      paymentModule: PaymentModule;
+      activityModule: ActivityModule;
+      networkModule: NetworkModule;
+      storageProvider: StorageProvider;
+      logger: Logger;
+    },
+  ) {}
+
+  createLease(agreement: Agreement, allocation: Allocation, options?: LeaseProcessOptions): LeaseProcess {
+    const paymentProcess = this.deps.paymentModule.createAgreementPaymentProcess(
+      agreement,
+      allocation,
+      options?.payment,
+    );
+    const lease = new LeaseProcess(
+      agreement,
+      this.deps.storageProvider,
+      paymentProcess,
+      this.deps.marketModule,
+      this.deps.activityModule,
+      this.deps.logger.child("lease-process"),
+      options,
+    );
+    paymentProcess.start();
+    lease.events.once("finalized", () => {
+      paymentProcess.stop();
+    });
```

Comment:
I just wonder if it is necessary to explicitly start and stop paymentProcess ... ? The PaymentProcess could automatically subscribe in constructor and unsubscribe is finished - when this.isFinished() === true